src/mgr.sol

```diff
@@ -186,124 +189,117 @@ contract TinlakeManager {
         operator = newOperator;
         emit SetOperator(newOperator);
     }
 
     function migrate(address dst) public auth  {
-        vat.hope(dst);
         dai.approve(dst, uint256(-1));
         gem.approve(dst, uint256(-1));
         live = false;
         emit Migrate(dst);
     }
 
     function file(bytes32 what, address data) public auth {
-        require(live, "TinlakeManager/not-live");
         emit File(what, data);
-        if (what == "vow") vow = data;
-        else if (what == "daiJoin") daiJoin = JoinLike(data);
+        if (what == "daiJoin") daiJoin = JoinLike(data);
         else if (what == "end")  end = EndLike(data);
+        else if (what == "urn") {
+            urn = MIP21UrnLike(data);
+            dai.approve(data, uint256(-1));
+        }
         else revert("Vat/file-unrecognized-param");
     }
 
     // --- Liquidation ---
+    // triggers a soft liquidation of the DROP collateral
+    // a redeemOrder is submitted to receive DAI back
     function tell() public auth {
         require(safe, "TinlakeManager/not-safe");
-        (uint256 ink, ) = vat.urns(ilk, address(this));
+        uint256 ink = gem.balanceOf(address(this));
         safe = false;
         gem.approve(tranche, ink);
         pool.redeemOrder(ink);
         emit Tell(ink);
     }
 
+    // triggers the payout of a DROP redemption
+    // method can be called multiple times after the liquidation until all
+    // DROP tokens are redeemed
     function unwind(uint256 endEpoch) public {
         require(!safe && glad && live, "TinlakeManager/not-soft-liquidation");
         (uint256 redeemed, , ,uint256 remainingDrop) = pool.disburse(endEpoch);
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(this));
-        uint256 dropReturned = sub(ink, remainingDrop);
-        require(dropReturned <= 2 ** 255, "TinlakeManager/overflow");
+        // here we use the urn instead of address(this)
+        (, uint256 art) = vat.urns(ilk, address(urn));
 
         (, uint256 rate, , ,) = vat.ilks(ilk);
         uint256 cdptab = mul(art, rate);
         uint256 payBack = min(redeemed, divup(cdptab, RAY));
 
-        daiJoin.join(address(this), payBack);
-        // Repay dai debt up to the full amount
-        // and exit the gems used up
-        uint256 dart = mul(RAY, payBack) / rate;
-        require(dart <= 2 ** 255, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this),
-                 0, -int256(dart));
-        vat.grab(ilk, address(this), address(this), address(this),
-                 -int256(dropReturned), 0);
-        vat.slip(ilk, address(this), -int256(dropReturned));
+        urn.wipe(payBack);
+
         // Return possible remainder to the owner
         dai.transfer(operator, dai.balanceOf(address(this)));
         emit Unwind(payBack);
     }
 
-    // --- Writeoff ---
+    // --- Write-off ---
+    // method should be called before RwaLiquidationOracle.cull()
```

Comment:
not sure what purpose this function plays anymore

Reply:
only to determine the `tab` at time of liquidation I suppose.

Reply:
Yes, that currently the only purpose. Do you see a more elegant way?

Reply:
There was some discussion around this in other channels. In short: Without touching MIP21 we can't make it anymore elegant.